### DISCUSSION TITLE
How to reset a store to its initial value?

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [ai/nanostores#138] [Q&A] [answered]
I'm new to Nano Stores,.  I'd like to use it on a complex modal.

When the modal closes, how would I reset all stores used by the modal to their respective initial values?

**Comment by ai** [comment, marked as answer]
`onMount(store, cb)` adds callback on the first mount and on unmount of latest subscriber.

```js
import { onMount } from 'nanostores'

onMount(store, () => {
  // first mount
  return () => {
    // unmount of latest subscriber
  }
})
```

Add it to all stores and reset their value. When you will close popup and there are not any other listeners for store’s data, this callback will be executed and reset the data in stores.
